Core/src/org/sleuthkit/autopsy/centralrepository/datamodel/EamArtifactUtil.java

```diff
@@ -65,101 +119,61 @@ public static String getEmailAddressAttrString() {
-                        value = artifactForInstance.getAttribute(new BlackboardAttribute.Type(BlackboardAttribute.ATTRIBUTE_TYPE.TSK_PHONE_NUMBER)).getValueString();
-                    } else if (null != artifactForInstance.getAttribute(new BlackboardAttribute.Type(BlackboardAttribute.ATTRIBUTE_TYPE.TSK_PHONE_NUMBER_FROM))) {
-                        value = artifactForInstance.getAttribute(new BlackboardAttribute.Type(BlackboardAttribute.ATTRIBUTE_TYPE.TSK_PHONE_NUMBER_FROM)).getValueString();
-                    } else if (null != artifactForInstance.getAttribute(new BlackboardAttribute.Type(BlackboardAttribute.ATTRIBUTE_TYPE.TSK_PHONE_NUMBER_TO))) {
-                        value = artifactForInstance.getAttribute(new BlackboardAttribute.Type(BlackboardAttribute.ATTRIBUTE_TYPE.TSK_PHONE_NUMBER_TO)).getValueString();
+            for (BlackboardAttribute attribute : artifact.getAttributes()) {
+                Integer customAttributeTypeId = attribute.getAttributeType().getTypeID();
+                if (CORRELATABLE_ATTRIBUTES.containsKey(customAttributeTypeId)) {
+                    addCorrelationAttributeToList(eamArtifacts, artifact,
+                            attribute,
+                            CORRELATABLE_ATTRIBUTES.get(customAttributeTypeId)
+                    );
+                } else {
+                    String attributeName = attribute.getAttributeType().getTypeName();
+
+                    if (attributeName == null) {
+                        continue;
                     }
-                    // Remove all non-numeric symbols to semi-normalize phone numbers, preserving leading "+" character
-                    if (value != null) {
-                        String newValue = value.replaceAll("\\D", "");
-                        if (value.startsWith("+")) {
-                            newValue = "+" + newValue;
-                        }
-                        value = newValue;
-                        // Only add the correlation attribute if the resulting phone number large enough to be of use
-                        // (these 3-5 digit numbers can be valid, but are not useful for correlation)
-                        if (value.length() > 5) {
-                            CorrelationAttributeInstance inst = makeCorrelationAttributeInstanceUsingTypeValue(artifactForInstance, EamDb.getInstance().getCorrelationTypeById(CorrelationAttributeInstance.PHONE_TYPE_ID), value);
-                            if (inst != null) {
-                                eamArtifacts.add(inst);
-                            }
-                        }
+
+                    if (attributeName.startsWith("TSK_PHONE_NUMBER_")) {
+                        addCorrelationAttributeToList(eamArtifacts, artifact,
+                                attribute,
+                                CorrelationAttributeInstance.PHONE_TYPE_ID
+                        );
```

Comment:
@bcarrier, are we going to add this sort of "secret handshake" on custom attribute names to the module developer documentation?

Reply:
We will leave this as is for now. A story will be created for defining how custom attributes should be defined to work with the CR.